src/OpenTelemetry.Instrumentation.MassTransit/Implementation/MassTransitDiagnosticListener.cs

```diff
@@ -61,6 +61,15 @@ public override void OnStartActivity(Activity activity, object payload)
 
             ActivityInstrumentationHelper.SetActivitySourceProperty(activity, ActivitySource);
             ActivityInstrumentationHelper.SetKindProperty(activity, this.GetActivityKind(activity));
+
+            try
+            {
+                this.options.Enrich?.Invoke(activity, "OnStartActivity", payload);
```

Comment:
we are moving to a new approach wherever feasible: https://github.com/open-telemetry/opentelemetry-dotnet/pull/3792

Please check this, and see if this can be done here too.

Reply:
@nicolasderycke Are you working on these suggestions?

Reply:
I will review these comments this week